mono/eglib/gqsort.c

```diff
@@ -40,14 +40,14 @@ typedef struct _QSortStack {
 	size_t count;
 } QSortStack;
 
-#define QSORT_PUSH(sp, a, c) (sp->array = a, sp->count = c, sp++)
+#define QSORT_PUSH(sp, a, c) (sp->array = g_cast (a), sp->count = c, sp++)
```

Comment:
the result type is known, can just use a normal explicit cast to `(char*)`

Reply:
g_cast becomes its own philosophy: no change to the C.

Reply:
But ok either way.

Reply:
if we really want no change to C, we should do:
```
#if defined (__cplusplus)
#define MONO_CXX_CAST(T,X) ((T)(X))
#else
#define MONO_CXX_CAST(T,X) (X)
#endif
```

Using `g_cast` in cases where we can use a simpler workaround seems unnecessary.

Reply:
‘g_cast’ is also more mechanical consisent easier to read write. And most of all, easy to find later & improve. Explicit casts will be harder to find. So maybe a separate macro as you show? Some of these probably ok indefinitely but some (not necessarily here) less so. Let me fix enum and sort and we can look at others later.

Reply:
I'm not against `MONO_CXX_CAST` but really we can probably add casts in C.